### ISSUE TITLE
systemctl reload-or-try-restart does not fail silently as per documentation

### ISSUE THREAD

**Issue by coling** [systemd/systemd#688]
The man page states:

```
   reload-or-try-restart PATTERN...
       Reload one or more units if they support it. If not, restart them instead.
       This does nothing if the units are not running. Note that, for compatibility
       with SysV init scripts, force-reload is equivalent to this command.
```

However:

```
[root@jimmy ~]# systemctl is-active httpd
inactive
[root@jimmy ~]# systemctl reload-or-try-restart httpd
Job for httpd.service invalid.
[root@jimmy ~]# echo $?
1
```

cf.

```
[root@jimmy ~]# systemctl try-restart httpd
[root@jimmy ~]# echo $?
0
```

So it's not behaving as described. This is with systemd 222

**Comment by poettering** [comment]
humm, that must be recent breakage. What's your systemd version?

**Comment by coling** [comment]
Sorry, was editing comments to include that! v222 here. But it's been like that for ages (at least 217). If it's not happening for you then it might be a downstream issue (in which case sorry!) but I don't _think_ any of our patches would touch that area of the code.

**Comment by coling** [comment]
(for reference [downstream Mageia bug](https://bugs.mageia.org/show_bug.cgi?id=15382)

**Comment by dvdhrm** [comment]
This is very easy to reproduce. Just write a random service file with `ExecReload=/bin/true`. If the unit is _NOT_ running, then `systemctl reload-or-try-restart foobar.service` fails with the given error code.

The underlying issue is `bus_unit_queue_job()` drops the `try-` information if the unit supports reloading. That is, if the unit supports reloading, the `reload-or-try-restart` (which is represented as `type == JOB_TRY_RESTART && reload_if_possible`) is folded into `JOB_RELOAD`. This clearly lacks the information that the call should not fail if the unit is not running.

I'm not sure how to fix this properly, and I cannot see how this ever worked. No patch from me, as I have really no clue how this is properly fixed. @poettering any ideas?

**Comment by poettering** [comment]
Hmm... the weird thing is that "reload-or-try-restart" is a bit of a misnomer... If it really is supposed to be silent if the unit supports reloading but isn't started, then we should have called it "try-reload-or-restart", i.e. the "try" prefix should cover the whole damn thing...

**Comment by poettering** [comment]
Fix waiting in #2464
